app/src/test/shared/components/Login/LoggedInUserBoxTests.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from "react";
+import { Sandbox } from "../../../Sandbox";
+import { expect } from "chai";
+import { shallow } from "enzyme";
+
+import { initialAuthState } from "../../../../main/shared/reducers/authReducer";
+import { InternalLink } from "../../../../main/shared/components/InternalLink";
+import { LoggedInUserBoxComponent } from "../../../../main/shared/components/Login/LoggedInUserBox";
+
+describe("LoggedInUserBoxComponent", () => {
+    const sandbox = new Sandbox();
+
+    afterEach(() => sandbox.restore());
+
+    it("renders nothing when the user is not logged in", () => {
+        const rendered = shallow(<LoggedInUserBoxComponent
+            loggedIn={initialAuthState.loggedIn}
+            username={initialAuthState.username}
+            logOut={()=>({})}
+        />);
+        expect(rendered.text()).to.be.empty;
+    });
+
+    it("renders log out link", () => {
+        const rendered = shallow(<LoggedInUserBoxComponent
+            loggedIn={true}
+            username="test.user"
+            logOut={()=>({})}
+        />);
+        expect(rendered.text()).to.contain("Logged in as test.user");
+        expect(rendered.find(InternalLink)).to.have.length(1);
+    });
+
+    it("clicking log out emits logOut event", () => {
+        const dispatchSpy = sandbox.createSpy();
```

Comment:
This test checks that clicking logout triggers the prop function `logOut`, but really we'd like to test that the `logOut` function is mapped to the right thing by `mapStateToProps`, and that `authActions.logOut()` is called.

Reply:
Note to self - this is now tested below :+1: